src/InstrumentationEngine/ExceptionSection.cpp

```diff
@@ -275,6 +275,8 @@ HRESULT MicrosoftInstrumentationEngine::CExceptionSection::AddNewExceptionClause
     IfFailRet(FindExceptionClauseInsertionPoint(pExceptionClause, iter));
     m_exceptionClauses.insert(iter, (IExceptionClause*)pExceptionClause);
 
+    *ppExceptionClause = pExceptionClause;
```

Comment:
Use *ppExceptionClause = pExceptionClaues.Detach() instead. Otherwise, this could cause an AV. The CComPtrReferencing the CExceptionClause will decrement the ref count. If the caller also uses a CComPtr, then it will also decrement the ref count when it is destroyed, causing the exception clause stored in m_exceptionClauses to become invalid.

Reply:
Sorry, I didn't think about the interaction with `CComPtr`, this is fixed now